compaction/compaction_manager.cc

```diff
@@ -886,12 +886,18 @@ future<> compaction_manager::perform_sstable_upgrade(database& db, column_family
 
 // Submit a column family to be scrubbed and wait for its termination.
 future<> compaction_manager::perform_sstable_scrub(column_family* cf, sstables::compaction_options::scrub::mode scrub_mode) {
+  // since we might potentially have ongoing compactions, and we
+  // must ensure that all sstables created before we run are scrubbed,
+  // we need to barrier out any previously running compaction.
     if (scrub_mode == sstables::compaction_options::scrub::mode::validate) {
         return perform_sstable_scrub_validate_mode(cf);
     }
+    return cf->run_with_compaction_disabled([this, cf, scrub_mode] {
```

Comment:
does this wait for all ongoing compactions to cease?

Reply:
yes.  this happens in `compaction_manager::remove(column_family* cf)`